### PR TITLE
Refactor liquidity params script.

### PR DESCRIPTION
**Pull request by talbaneth**
Now get json file and supports several options.

### CODE REVIEW

File: scripts/get_liquidity_params.py [hunk 1]
```diff
@@ -1,170 +1,200 @@
...
+        ether,
+        tokens
+):
+
+    max_rate = price / (1 - rate * price * tokens)
+    min_rate = price / np.exp(rate * ether)
+    min_ratio = min_rate / price
+    max_ratio = max_rate / price
+    print("min_min_ratio: " + str(min_ratio))
+    print("max_max_ratio: " + str(max_ratio) + "\n")
```
Comment (yaronvel) [inline scripts/get_liquidity_params.py]: do you really need `\n` here? if so, why not in the other places

File: scripts/get_liquidity_params.py [hunk 2]
```diff
@@ -1,170 +1,200 @@
...
+    _rInFp = rate * (2 ** num_formula_precision_bits)
+    _pMinInFp = min_rate * (2 ** num_formula_precision_bits)
+    _numFpBits = num_formula_precision_bits
+    _maxCapBuyInWei = max_cap_buy_eth * (10 ** 18)
+    _maxCapSellInWei = max_cap_sell_eth * (10 ** 18)
+    _feeInBps = fee_in_precents * 100
+    _maxTokenToEthRateInPrecision = max_rate * (10 ** 18)
+    _minTokenToEthRateInPrecision = min_rate * (10 ** 18)
+
+    print("_rInFp: %.0f" % _rInFp)
```
Comment (yaronvel) [inline scripts/get_liquidity_params.py]: technically, everything here should be in int, not float (%d)

File: scripts/get_liquidity_params.py [hunk 3]
```diff
@@ -1,170 +1,200 @@
...
+    return min_ratio, max_ratio
+
+
+def calc_rate_from_eth_balance(ether, price, min_ratio):
+    min_rate = min_ratio * price
+    rate = (1 / ether) * np.log(price / min_rate)
+    print("rate: " + str(rate))
+
+
+def calc_params_from_ratios(
```
Comment (yaronvel) [inline scripts/get_liquidity_params.py]: think we might want to offer also a simpler function that will set some defaults. maybe extract `r` from initial balances etc.

File: scripts/get_liquidity_params.py [hunk 4]
```diff
@@ -1,170 +1,200 @@
...
+                "max_ratio",
+                "ether",
+                "tokens"
+            ],
+            d.keys()
+        )
+
+        (min_min_ratio, max_max_ratio) = \
+            calc_min_max_ratio_from_balances(
+                d["price"],
```
Comment (yaronvel) [inline scripts/get_liquidity_params.py]: use amit ticks. to match input and input params

File: scripts/get_liquidity_params.py [hunk 5]
```diff
@@ -1,170 +1,200 @@
...
+
+
+def calc_balances_from_pmin_pmax(
+    rate,
+    price,
+    min_ratio,
+    max_ratio
+):
+
+    min_rate = min_ratio * price
```
[the remark this answers is not in the document]
Comment (talbaneth) [reply scripts/get_liquidity_params.py]: done.

File: scripts/get_liquidity_params.py [hunk 6]
```diff
@@ -1,170 +1,200 @@
...
+    max_rate = max_ratio * price
```
[the remark this answers is not in the document]
Comment (talbaneth) [reply scripts/get_liquidity_params.py]: done.

File: scripts/get_liquidity_params.py [hunk 7]
```diff
@@ -1,170 +1,200 @@
...
-        #    min_token_to_eth_rate=option["min_token_to_eth_ratio"] * token_options["p0"],
-        #    max_token_to_eth_rate=option["max_token_to_eth_ratio"] * token_options["p0"]
-        #)
-
-        #liq.calc_params_based_on_ratios(option["min_token_to_eth_ratio"],
-        #                                option["max_token_to_eth_ratio"])
+from pprint import pprint
+
+
+def calc_balances_from_pmin_pmax(
```
[the remark this answers is not in the document]
Comment (talbaneth) [reply scripts/get_liquidity_params.py]: done.

File: scripts/get_liquidity_params.py [hunk 8]
```diff
@@ -1,170 +1,200 @@
...
+    print("tokens_in_eth: " + str(tokens_in_eth))
+
+    return ether, tokens
+
+
+def calc_min_max_ratio_from_balances(
+        price,
+        rate,
+        ether,
+        tokens
```
[the remark this answers is not in the document]
Comment (talbaneth) [reply scripts/get_liquidity_params.py]: changed to ether_amount and token_amount

File: scripts/get_liquidity_params.py (hunk 1 above)
Comment (talbaneth) [reply scripts/get_liquidity_params.py]: not really, just wanted a separation between the above prints and the actual liquidity params print. 
this is the only place where we print something and than another function prints something else right away.
anw I removed the \n.

File: scripts/get_liquidity_params.py (hunk 2 above)
Comment (talbaneth) [reply scripts/get_liquidity_params.py]: yeah you are right, changed to %d.

File: scripts/liquidity_input_params.json [hunk 9]
```diff
@@ -0,0 +1,12 @@
+{
+  "liquidity_rate": 0.01,
+  "initial_ether_amount": 70.0,
+  "initial_token_amount": 724638,
+  "initial_price":  0.000069,
+  "min_supported_price_factor": 0.5,
+  "max_supported_price_factor" : 2.0,
+  "max_tx_buy_amount_eth": 10.0,
+  "max_tx_sell_amount_eth": 10.0,
+  "fee_in_precents": 0.25,
```
Comment (aenaus) [inline scripts/liquidity_input_params.json]:
```suggestion
  "fee_in_percent": 0.25,
```

File: scripts/get_liquidity_params.py [hunk 10]
```diff
@@ -1,170 +1,272 @@
...
+            " by " +  str(diff_percent) + "%"
+        )
+
+def print_params(
+    liquidity_rate,
+    initial_price,
+    formula_precision_bits,
+    max_tx_buy_amount_eth,
+    max_tx_sell_amount_eth,
+    fee_in_precents,
```
Comment (aenaus) [inline scripts/get_liquidity_params.py]:
```suggestion
    fee_in_percent,
```

File: scripts/get_liquidity_params.py [hunk 11]
```diff
@@ -1,170 +1,272 @@
...
+
+    max_supported_price = max_supported_price_factor * initial_price
+    min_supported_price = min_supported_price_factor * initial_price
+
+    _rInFp = liquidity_rate * (2 ** formula_precision_bits)
+    _pMinInFp = min_supported_price * (2 ** formula_precision_bits)
+    _numFpBits = formula_precision_bits
+    _maxCapBuyInWei = max_tx_buy_amount_eth * (10 ** 18)
+    _maxCapSellInWei = max_tx_sell_amount_eth * (10 ** 18)
+    _feeInBps = fee_in_precents * 100
```
Comment (aenaus) [inline scripts/get_liquidity_params.py]:
```suggestion
    _feeInBps = fee_in_percent * 100
```

File: scripts/get_liquidity_params.py [hunk 12]
```diff
@@ -1,170 +1,272 @@
...
+    d = json.load(json_data)
+
+    if args.get == 'params':
+        require_args([
+                "liquidity_rate",
+                "initial_price",
+                "formula_precision_bits",
+                "max_tx_buy_amount_eth",
+                "max_tx_sell_amount_eth",
+                "fee_in_precents",
```
Comment (aenaus) [inline scripts/get_liquidity_params.py]:
```suggestion
                "fee_in_percent",
```

File: scripts/get_liquidity_params.py [hunk 13]
```diff
@@ -1,170 +1,272 @@
...
+            max_supported_price_factor=d["max_supported_price_factor"]
+        )
+
+        print_params(
+            liquidity_rate=d["liquidity_rate"],
+            initial_price=d["initial_price"],
+            formula_precision_bits=d["formula_precision_bits"],
+            max_tx_buy_amount_eth=d["max_tx_buy_amount_eth"],
+            max_tx_sell_amount_eth=d["max_tx_sell_amount_eth"],
+            fee_in_precents=d["fee_in_precents"],
```
Comment (aenaus) [inline scripts/get_liquidity_params.py]:
```suggestion
            fee_in_percent=d["fee_in_percent"],
```

File: scripts/liquidity_input_params.json (hunk 9 above)
Comment (talbaneth) [reply scripts/liquidity_input_params.json]: change to fee_percent

File: scripts/get_liquidity_params.py (hunk 10 above)
Comment (talbaneth) [reply scripts/get_liquidity_params.py]: change to fee_percent

File: scripts/get_liquidity_params.py (hunk 11 above)
Comment (talbaneth) [reply scripts/get_liquidity_params.py]: change to fee_percent

File: scripts/get_liquidity_params.py (hunk 12 above)
Comment (talbaneth) [reply scripts/get_liquidity_params.py]: change to fee_percent

File: scripts/get_liquidity_params.py (hunk 13 above)
Comment (talbaneth) [reply scripts/get_liquidity_params.py]: change to fee_percent

**Comment by talbaneth** [comment]
done SPyrus requested one letter change,

**Pull request merged**